app/Services/VoucherEvaluator/Evaluations/FamilyHasUnverifiedChildren.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Services\VoucherEvaluator\Evaluations;
+
+use App\Specifications\IsVerified;
+use Carbon\Carbon;
+
+class FamilyHasUnverifiedChildren extends BaseFamilyEvaluation
+{
+    public $reason = 'has children needing ID';
+    private $specification;
+    /**
+     * FamilyIsPregnant constructor.
+     * @param Carbon|null $offsetDate
+     */
+    public function __construct(Carbon $offsetDate = null)
+    {
+        parent::__construct($offsetDate);
+
+        // Pregnancies or under school age.
+        $this->specification = new IsVerified();
+    }
+
+    /**
+     * @param $candidate
+     * @return FamilyHasNoEligibleChildren|void|null
+     */
+    public function test($candidate)
+    {
+        parent::test($candidate);
+
+        $children = $candidate->children->all();
+
+        $satisfiers = array_filter(
+            $children,
+            function ($child) {
+                // If we satisfy the rule
+                return ($this->specification->isSatisfiedBy($child));
+            }
+        );
+
+        if (empty($satisfiers)) {
```

Comment:
I keep seeing people checking for empty arrays with the `$empty()` or the `count($array) < 1` or `isset()`. Are there any advantages of one way over another? Or it's just down to personal preference? I'll look deeper into it - I did already but my memory is not that good - but am also curious about your choice.

Reply:
isset() will ask if the variable is null, _or even declared_. It exists because you sometimes deep objects and array structures are lazily populated.

empty is a language construct, not really a function. It's a bit more flexible than isset(), in that IIRC, it can spot that an array *exists*, but has no content - see https://www.php.net/manual/en/function.empty for the difference.

count($array) assumes there _is_ an array to count.

I use empty becuase i'm a little lazy, and it seems to catch conditions that i would define as "not present, or not useful"